### DISCUSSION TITLE
something recreates hetzner loadbalancer

### DISCUSSION THREAD

**Discussion by rainoko** [kube-hetzner/terraform-hcloud-kube-hetzner#126] [Q&A] [answered]
Seems that something recreates job helm-install-traefik.
This recreates new LB in hetzner and new IP. This is of course problem when using DNS records.

What can cause this recreation of helm-install-job?

**Comment by mysticaltech** [comment]
What happens is that every time you add or remove a node, the Hetzner CCM recreates the LB. 

So best way is just to have external-dns in place to automatically do the adjustment in the event of an IP change.

**Reply by rainoko** [reply]
Interesting part is that I haven't t created or removed node or enything else in the cluster.
LB recreation is happening randomly and usually at night. So I haven't figured out yet what causes it.

**Reply by mnencia** [reply]
I've seen it once in the past when i manually update the traefik chart. I'm afraid this happens when k3s update.

**Reply by phaer** [reply]
@rainoko the CCM logs should tell you when exactly it re-creates the LB. Maybe you could see whether timestamps there match timestamps of the last upgrade?

**Comment by phaer** [comment, marked as answer]
I believe this might be solved by https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/135

Can you confirm @rainoko?
